drivers/note/noteram_driver.c

```diff
@@ -413,12 +564,22 @@ void sched_note_add(FAR const void *note, size_t notelen)
   FAR const char *buf = note;
   unsigned int head;
   unsigned int next;
+  irqstate_t flags;
 
+  flags = up_irq_save();
 #ifdef CONFIG_SMP
-  irqstate_t flags = up_irq_save();
   spin_lock_wo_note(&g_noteram_lock);
 #endif
 
+  if (g_noteram_info.ni_overwrite == NOTERAM_MODE_OVERWRITE_OVERFLOW)
```

Comment:
I have also fixed here.
- ni_overwrite flag must be checked in the locked region.
- up_irq_save() is needed in not SMP case.